### ISSUE TITLE
Root cannot use IPPROTO_ICMP by default

### ISSUE THREAD

**Issue by andrewthad** [andrewthad/ping#7]
Quoting https://lwn.net/Articles/443051/:

> socket(2) is restricted to the group range specified in "/proc/sys/net/ipv4/ping_group_range".  It is "1 0" by default, meaning that nobody (not even root) may create ping sockets.

This means that the package description for `ping` is currently wrong. Running as root is not an option.

**Comment by chessai** [comment]
the first option worked great for me.

**Comment by chessai** [comment]
I think the advice of giving the executable CAP_NET_RAW is probably the most advisable due to security concerns.

**Comment by andrewthad** [comment]
I'm not aware of any security concerns with `ping_group_range`. There's no way to escalate privileges with it.

**Comment by chessai** [comment]
Yeah but now you're giving potentially too many groups the ability to do icmp as non root

**Comment by andrewthad** [comment]
Is that a security concern? In linux, by default, every user can open TCP and UDP sockets to peers on the network. How is allowing users to make ICMP echo requests any worse? Also, the docs for `ping` don't currently advocate opening it up to every user on the system. It's possible to set `ping_group_range` to something like `1001 1001` to lock it down to a single user. The description should be more fleshed out. It currently reads:

> The kernel parameter net.ipv4.ping_group_range has been configured to allow pings to all IP addresses

This should really talk about the three common ways to use this: only root, only one non-root user, and every user.
